hw/ip/pinmux/doc/_index.md

```diff
@@ -6,122 +6,387 @@ title: "Pinmux Technical Specification"
 
-The pin multiplexor module intends to give maximum flexibility of peripheral and chip wiring to the software running on the device.
-The assumption is that the wiring is done once at the initialization of the application based upon usage of the device within the broader system.
-How this wiring is most effectively done is outside the scope of this document, but a section below briefly discusses use cases.
+The wakeup detectors can detect patterns such as rising / falling edges and pulses of a certain width up to 255 AON clock cycles.
+Each wakeup detector can listen on any one of the MIO / DIO signals that are routed through the `pinmux`, and if a patterns is detected, the power manager is informed of that event via a wakeup request.
 
-## Block Diagram
+The `pinmux` module itself is in the always-on (AON) power domain, and as such does not loose configuration state when a deep sleep power cycle is performed.
+However, only the wakeup detector logic will be actively clocked during deep sleep in order to save power.
 
-The diagram below shows connectivity between four arbitrary chip pins, named `MIO_00` .. `MIO_03`, and several peripheral inputs and outputs.
-This shows the connectivity available in all directions, as well as the control registers described later in this document.
-Two example peripherals (`uart` and `spidev`) are attached to the pinmux in this example, one with one input and one output, the other with three inputs and one output.
-The diagram also shows the `padring` module (from the `padctrl` IP) which instantiates the bidirectional chip pads with output enable control.
+See the [retention logic]({{< relref "#retention-logic">}}) and [wakeup detectors]({{< relref "#wakeup-detectors">}}) sections for more details about the mux implementation.
 
-![Pinmux Block Diagram](pinmux_block_diagram.svg)
+### Test and Debug Access
 
-Note that apart from selecting a specific input pad, the `periph_insel[*]` signals can also be used to tie the peripheral input to 0 or 1.
-Likewise, the output select signals `mio_outsel[*]` can also be used to constantly drive an output pin to 0/1 or to put it into high-Z state (default).
-The output enable and the associated data signal (i.e. `periph_to_mio` and `periph_to_mio_oe`) are indexed with the same select signal to allow the peripheral hardware to determine the pad direction instead of demoting that control to software.
+The hardware strap sampling and TAP isolation logic provides test and debug access to the chip during specific life cycle states.
+This mechanism is explained in more detail in the [strap sampling and TAP isolation]({{< relref "#strap-sampling-and-tap-isolation">}}) section.
 
-Additional details about the signal names and parameters are given in the sections below.
+### Pad Attributes
+
+Additional pad-specific features such as inversion, pull-up, pull-down, virtual open-drain, drive-strength, input/output inversion etc. can be exercise via the pad attribute CSRs.
```

Comment:
In previous text, i saw `open drain` and `drive-strength` without the `-`. If not too much work, maybe better to unify them?

Reply:
oh yeah, let's unify them.